File: pi/go/MonteCarlo.go
package main

import (
	"fmt"
	"math"
	"math/rand"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())

	x, total, p, l, a := 0., 0, 0., 1., 0.

	for l > 1e-9 {
		total++

		x += math.Pow(1-math.Pow(rand.Float64(), 2), .5)

		p = 4 * float64(x) / float64(total)
		l = math.Abs(math.Pi - p)
		a = 100 - (math.Abs(math.Pi-p)*100)/math.Pi

		if total%100000 == 0 {
			fmt.Printf("Iteration: %d\tEstimate: %.8f\tLoss: %e\tAccuracy: %.8f\n", total, p, l, a)
		}
	}
	fmt.Printf("Iteration: %d\tEstimate: %.8f\tLoss: %e\tAccuracy: %.8f\n", total, p, l, a)

	fmt.Scanln()
}

File: pi/python/MonteCarlo.py
import math
import random


text = ""
x, total, l = 0, 0, 1

while l > 1e-9:
    total += 1

    x += (1 - (random.uniform(0, 1)**2))**.5

    p = 4 * x / total
    l = abs(math.pi - p)
    a = 100 - (abs(math.pi - p) * 100) / math.pi

    if not total % 100000:
        print(f"Iteration: {total}\tEstimate: {round(p, 8)}\tLoss: {l}\tAccuracy: {round(a, 8)}")

print(f"Iteration: {total}\tEstimate: {round(p, 8)}\tLoss: {l}\tAccuracy: {round(a, 8)}")
